osu.Game/Screens/Select/Carousel/CarouselGroupEagerSelect.cs

```diff
@@ -101,8 +101,12 @@ private void attemptSelection()
 
         protected virtual CarouselItem GetNextToSelect()
         {
-            return Children.Skip(lastSelectedIndex).FirstOrDefault(i => !i.Filtered.Value) ??
-                   Children.Reverse().Skip(InternalChildren.Count - lastSelectedIndex).FirstOrDefault(i => !i.Filtered.Value);
+            var forwards = Children.Skip(lastSelectedIndex).FirstOrDefault(i => !i.Filtered.Value);
```

Comment:
This method needs documentation via comments. Also potential consideration to the fact that by resolving these (via `FirstOrDefault`) ahead of time, you are incurring a much higher performance overhead, which may be important since this is operating on all beatmaps in the carousel. The previous code was not doing this.

Reply:
Yeah, but if we want carousel to move in both directions then resolving ahead of time is still O(n) instead of sorting that I had here few commits before which is O(nlogn). 

And I think this is also fine because carousel root only calls this eagerly, that is when it loses selection which can only occur on filter changes.